view/handlers/comment.go

```diff
@@ -0,0 +1,226 @@
+		http.Error(w, "cannot get created comment", http.StatusNotFound)
+
+		return
+	}
+
+	out := viewmodels.GetCommentResponse{
+		ID:        comment.ID,
+		Content:   comment.Content,
+		CreatedAt: comment.CreatedAt,
+		CreatedBy: comment.CreatedBy,
+		PostID:    comment.PostID,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err = json.NewEncoder(w).Encode(out); err != nil {
+		http.Error(w, "cannot encode data into JSON", http.StatusInternalServerError)
+
+		return
+	}
+}
+
+func (h *Handlers) GetComment(w http.ResponseWriter, r *http.Request) {
+	commentID := mux.Vars(r)["id"]
+
+	comment, err := h.service.GetComment(r.Context(), commentID)
+	if err != nil {
+		http.Error(w, "cannot get comment with such ID", http.StatusNotFound)
+
+		return
+	}
+
+	out := viewmodels.GetCommentResponse{
+		ID:        comment.ID,
+		Content:   comment.Content,
+		CreatedAt: comment.CreatedAt,
+		CreatedBy: comment.CreatedBy,
+		PostID:    comment.PostID,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err = json.NewEncoder(w).Encode(out); err != nil {
+		http.Error(w, "cannot encode data into JSON", http.StatusInternalServerError)
+
+		return
+	}
+}
+
+func (h *Handlers) UpdateComment(w http.ResponseWriter, r *http.Request) {
+	commentID := mux.Vars(r)["id"]
+
+	comment, err := h.service.GetComment(r.Context(), commentID)
+	if err != nil {
+		http.Error(w, "cannot get comment with such ID", http.StatusNotFound)
+
+		return
+	}
+
+	in := viewmodels.UpdateCommentRequest{
+		Content: comment.Content,
+	}
+
+	if err = json.NewDecoder(r.Body).Decode(&in); err != nil {
+		http.Error(w, "cannot decode data from JSON", http.StatusBadRequest)
+
+		return
+	}
+
+	err = h.service.UpdateComment(r.Context(), models.UpdateCommentPayload{
+		CommentID: commentID,
+		Content:   in.Content,
+	})
+	if err != nil {
+		http.Error(w, "cannot update comment", http.StatusBadRequest)
+
+		return
+	}
+
+	comment, err = h.service.GetComment(r.Context(), commentID)
+	if err != nil {
+		http.Error(w, "cannot get updated comment", http.StatusInternalServerError)
+
+		return
+	}
+
+	out := viewmodels.GetCommentResponse{
+		ID:        comment.ID,
+		Content:   comment.Content,
+		CreatedAt: comment.CreatedAt,
+		CreatedBy: comment.CreatedBy,
+		PostID:    comment.PostID,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err = json.NewEncoder(w).Encode(out); err != nil {
+		http.Error(w, "cannot encode data into JSON", http.StatusInternalServerError)
+
+		return
+	}
+}
+
+func (h *Handlers) DeleteComment(w http.ResponseWriter, r *http.Request) {
+	commentID := mux.Vars(r)["id"]
+
+	if err := h.service.DeleteComment(r.Context(), commentID); err != nil {
+		http.Error(w, "cannot delete comment with such ID", http.StatusNotFound)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handlers) GetListOfComments(w http.ResponseWriter, r *http.Request) {
+	vars := r.URL.Query()
+	filterByField := vars.Get("filterByField")
```

Comment:
get parameters usually should be written in lower case, for instance `filter-field`, `filter-value` and so on